src/token/fedauth-info-parser.js

```diff
@@ -0,0 +1,61 @@
+const async = require('async');
+
+function readFedAuthInfoOpt(data, infoData, FEDAUTHINFOID, callback) {
```

Comment:
It'd be great if this function could be refactored a bit for better readability. It can be a regular non-async method, and I think it should return a `{ stsurl: '...', spn: '...' }` object.

Having it take and modify the `infoData` object in place is a bit confusing. 🤔

Reply:
Also, `FEDAUTHINFOID` could be a top-level constant, instead of being passed in as well.